### DISCUSSION TITLE
Can the keycloak adapters work with Java 17?

### DISCUSSION THREAD

**Discussion by kkcmadhu** [keycloak/keycloak#17207] [Q&A]
The keycloak 20 release notes says. 

**"Keycloak now supports OpenJDK 17 both for the server and adapters."**

Does that mean the existing adapters like java servlet oidc adapter (https://github.com/keycloak/keycloak/blob/main/adapters/oidc/servlet-filter/) work with java 17??

i still see these code using javax.servlet instead of jakarta.servlet in keycloak 20.0.5 release https://github.com/keycloak/keycloak/blob/20.0.5/adapters/oidc/servlet-filter/src/main/java/org/keycloak/adapters/servlet/KeycloakOIDCFilter.java

The documentation and messaging are pretty confusing.

Can we have a clear answer to these?

1. What is happening to the adapter libraires, are they are being deprecated if so? I don't really see any @deprecated annotation in source code or anywhere else.

3. What is the alternative libraries keycloak plans to propose?
 
5. How will k_  events, which are part of keycloak spec and not oidc be dealt with?
 
7. What do we mean  by keycloak 20  server and adapters are is java 17 compatible, when the code still uses javax.servelt (instead of jakarta.servlet)?
 
9. If the adapter will be dropped, is there any proposal to make them community maintained, and if so, is there a separate repository being created for the same?

There are many hanging questions, without a proper communication/plan or documentation.

Blogs/ roadmaps says one thing(https://www.keycloak.org/2022/03/releases.html)
Release Notes suggests something differnt (https://www.keycloak.org/2022/11/keycloak-2000-released.html)
The code actually does not really match with roadmap or release note E.g. https://github.com/keycloak/keycloak/blob/20.0.5/adapters/oidc/servlet-filter/src/main/java/org/keycloak/adapters/servlet/KeycloakOIDCFilter.java )

It will be really useful and help the community if a clear plan/direction/migration path is laid out and communicated with regards to these questions.

**Comment by kkcmadhu** [comment]
keycloak 21 too uses javax.servlet pakcages,
While i see that in most other places  (server code), the javax is replaced with jakarta.servlet.

I am not sure what we mean by 
"Adapters remain supported on Java 8, Java 11, and Java 17. However, we are planning to remove support for Java 8 in the not too distant future." in  keycloak 21 release notes. https://www.keycloak.org/2023/02/keycloak-2100-released

can some one please explain.

**Reply by kkcmadhu** [reply]
i am still confused about this, it will be great if keycloak team can throw some light on this.

**Comment by stianst** [comment]
You are mixing two things here. Java 17 and Jakarta EE 10 are not the same thing. In terms of Jakarta EE 10 we are not planning to support adapters for that, I would suggest taking a look at Java Security 3.0 which brings OIDC support to Jakarta. 

Please see https://www.keycloak.org/2023/03/adapter-deprecation-update.html
